### PR TITLE
ERS-439 remove size distribution classcode table

### PR DESCRIPTION
**Pull request by RobinMilding**
Two tables have been removed and their contents added to four new columns in activity_fa_catch.

### CODE REVIEW

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java [hunk 1]
```diff
@@ -180,6 +184,60 @@ protected VesselTransportMeansEntity getVesselTransportMeansForRelocation(FaCatc
...
+    protected SizeDistribution getSizeDistribution(FaCatchEntity faCatch) {
+        if (faCatch == null) {
+            return null;
+        }
+
+        SizeDistribution result = new SizeDistribution();
+        if (faCatch.getSizeDistributionCategoryCode() != null && faCatch.getSizeDistributionCategoryCodeListId() != null) {
+            CodeType code = new CodeType();
+            code.setValue(faCatch.getSizeDistributionCategoryCode());
+            code.setListID(faCatch.getSizeDistributionCategoryCodeListId());
```
Comment (psjobergprivat) [inline service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java:195]: Missing a row with result.setCategoryCodes ... code ... here, right?

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java [hunk 2]
```diff
@@ -180,6 +184,60 @@ protected VesselTransportMeansEntity getVesselTransportMeansForRelocation(FaCatc
...
+
+        return sizeDistribution.getCategoryCode().getListID();
+    }
+
+    protected String getSizeDistributionClassCode(SizeDistribution sizeDistribution) {
+        if (sizeDistribution == null || sizeDistribution.getClassCodes() == null || sizeDistribution.getClassCodes().isEmpty()) {
+            return null;
+        }
+
+        return sizeDistribution.getClassCodes().get(0).getValue();
```
Comment (psjobergprivat) [inline service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java:230]: We should get the approach confirmed by Andreas, but both Robert and me has written code where we throw an IllegalArgumentException if we get a list with >1 item. The exception description should contain info about what happened, and what values were involved so it is easy to find the spot in the source (eg flux document) where this deviation from the Flux doc happened.

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java [hunk 3]
```diff
@@ -536,15 +534,19 @@ private void getQuantityFromAapProduct(FaCatchEntity faCatch, Double quantity) {
...
+                if (specifiedSizeDistribution.getCategoryCode() != null) {
+                    faCatchEntity.setSizeDistributionCategoryCode(specifiedSizeDistribution.getCategoryCode().getValue());
+                    faCatchEntity.setSizeDistributionCategoryCodeListId(specifiedSizeDistribution.getCategoryCode().getListID());
                 }
 
-                faCatchEntity.setSizeDistribution(sizeDistributionEntity);
+                List<CodeType> classCodes = specifiedSizeDistribution.getClassCodes();
+                if (!CollectionUtils.isEmpty(classCodes)) {
+                    if (classCodes.size() > 1) {
+                        throw new IllegalArgumentException("Failed to map FACatch.specifiedSizeDistribution, more than one SizeDistributionType.ClassCode found");
```
Comment (psjobergprivat) [inline service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java]: Ah yes, this is it! I think it is very helpful if the values from the involved classCodes are provided in the description.

**Review by psjobergprivat — COMMENTED** [review]
I think I found a bug

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java (hunk 1 above)
Comment (RobinMilding) [reply service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java:195]: So embarrassing, you are completely right. Good thing you are a thorough reviewer!

File: service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java [hunk 4]
```diff
@@ -536,15 +535,20 @@ private void getQuantityFromAapProduct(FaCatchEntity faCatch, Double quantity) {
...
+                if (specifiedSizeDistribution.getCategoryCode() != null) {
+                    faCatchEntity.setSizeDistributionCategoryCode(specifiedSizeDistribution.getCategoryCode().getValue());
+                    faCatchEntity.setSizeDistributionCategoryCodeListId(specifiedSizeDistribution.getCategoryCode().getListID());
                 }
 
-                faCatchEntity.setSizeDistribution(sizeDistributionEntity);
+                List<CodeType> classCodes = specifiedSizeDistribution.getClassCodes();
+                if (!CollectionUtils.isEmpty(classCodes)) {
+                    if (classCodes.size() > 1) {
+                        String values = classCodes.stream().map(CodeType::getValue).collect(Collectors.joining(", "));
```
Comment (psjobergprivat) [inline service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FishingActivityMapper.java:546]: I like! Troubleshooting will be a piece of cake

**Review by psjobergprivat — APPROVED** [review]

**Pull request merged**